### PR TITLE
Add SwitchCaseAlignmentRule

### PR DESCRIPTION
**Pull request by austinlu**
This finds all switch substructures and checks that all case and default statements nested within are vertically aligned

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,9 @@
 * Add `fallthrough` rule that flags usage of `fallthrough`.  
   [Marcelo Fabri](https://github.com/marcelofabri)
   [#1834](https://github.com/realm/SwiftLint/issues/1834)
+* Add `switch_case_alignment` opt-in rule to `check` case and `default`
```
Comment (jpsim) [inline CHANGELOG.md]: Single newline between changelog entries.

File: Source/SwiftLintFramework/Rules/SwitchCaseAlignmentRule.swift [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+//  SwiftLint
+//
+//  Created by Austin Lu on 9/6/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct SwitchCaseAlignmentRule: ASTRule, OptInRule, ConfigurationProviderRule {
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/SwitchCaseAlignmentRule.swift]: Why is this marked as opt-in? From what I can tell, the community has (mostly) settled on `case` being vertically aligned with `switch` following Xcode's default formatting.

File: Source/SwiftLintFramework/Rules/SwitchCaseAlignmentRule.swift [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "switch_case_alignment",
+        name: "Switch and Case Statement Alignment",
+        description: "Case statements should vertically align with the enclosing switch statement itself.",
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/SwitchCaseAlignmentRule.swift]: > Case statements should vertically align with its enclosing switch statement.

File: CHANGELOG.md (hunk 1 above)
Comment (austinlu) [reply CHANGELOG.md]: Updated, thanks!

**Comment by jpsim** [comment]
At first glance, this looks excellent. Very clean implementation. I'm curious to see if OSSCheck catches anything.

File: Source/SwiftLintFramework/Rules/SwitchCaseAlignmentRule.swift (hunk 2 above)
Comment (austinlu) [reply Source/SwiftLintFramework/Rules/SwitchCaseAlignmentRule.swift]: I agree this rule shouldn't be controversial, and I'm happy to make it enabled by default. PR updated!

File: Source/SwiftLintFramework/Rules/SwitchCaseAlignmentRule.swift (hunk 3 above)
Comment (austinlu) [reply Source/SwiftLintFramework/Rules/SwitchCaseAlignmentRule.swift]: Done, thanks!

**Comment by jpsim** [comment]
Thanks for the PR! Merged in b2a17026ccd27456a93ed38bddcd40872f8df732...59ff3366458d2797fc0454d9733362100be7d343

**Pull request closed without merging**